Specs/Scene/Cesium3DTilesetSpec.js

```diff
@@ -0,0 +1,633 @@
+            tileset.loadProgress.addEventListener(function(numberOfPendingRequests, numberProcessing) {
+                var expected = results[i++];
+                expect(numberOfPendingRequests).toEqual(expected[0]);
+                expect(numberProcessing).toEqual(expected[1]);
+            });
+            viewRootOnly();
+            scene.renderForSpecs();
+            return Cesium3DTilesTester.waitForPendingRequests(scene, tileset);
+        });
+    });
+
+    it('tile visible event is raised', function() {
+        viewRootOnly();
+        return Cesium3DTilesTester.loadTileset(scene, tilesetUrl).then(function(tileset) {
+            tileset.tileVisible.addEventListener(function(tile) {
+                expect(tile).toBe(tileset._root);
+                expect(tileset._root.visibility(scene.frameState.cullingVolume)).not.toEqual(CullingVolume.MASK_OUTSIDE);
+            });
+
+            scene.renderForSpecs();
+        });
+    });
+
+    it('destroys', function() {
+        return Cesium3DTilesTester.loadTileset(scene, tilesetUrl).then(function(tileset) {
+            var root = tileset._root;
+            expect(tileset.isDestroyed()).toEqual(false);
+            scene.primitives.remove(tileset);
+            expect(tileset.isDestroyed()).toEqual(true);
+
+            // Check that all tiles are destroyed
+            expect(root.isDestroyed()).toEqual(true);
+            expect(root.children[0].isDestroyed()).toEqual(true);
+            expect(root.children[1].isDestroyed()).toEqual(true);
+            expect(root.children[2].isDestroyed()).toEqual(true);
+            expect(root.children[3].isDestroyed()).toEqual(true);
+        });
+    });
+
+    it('destroys before loadTilesJson finishes', function() {
+        var tileset = scene.primitives.add(new Cesium3DTileset({
+            url : tilesetUrl
+        }));
+        scene.primitives.remove(tileset);
+        // TODO : hard to test this one (check that promise never resolves or rejects?)
+        // Should destroy function reject ready promise?
+    });
+
+    it('destroys before external tiles.json finishes loading', function() {
+        viewNothing();
+        return Cesium3DTilesTester.loadTileset(scene, tilesetOfTilesetsUrl).then(function(tileset) {
+            var root = tileset._root;
+
+            viewRootOnly();
+            scene.renderForSpecs(); // Request external tiles.json
+
+            var stats = tileset._statistics;
+            expect(stats.numberOfPendingRequests).toEqual(1);
+            scene.primitives.remove(tileset);
+
+            return root.readyPromise.then(function(root) {
+                fail('should not resolve');
+            }).otherwise(function(error) {
+                // Expect the root to not have added any children from the external tiles.json
+                expect(root.children.length).toEqual(0);
+                // TODO : check that request scheduler is empty
```

Comment:
I may just leave them there for now and wait until the request scheduler is ready.